samples/matter/common/src/binding_handler.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "binding_handler.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::app;
+
+CHIP_ERROR BindingHandler::Init(DeviceChangedHandler handler)
+{
+	VerifyOrReturnError(mDeviceChangedHandler != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
I think you should check `handler` not `mDeviceChangedHandler`. Also where is `mDeviceChangedHandler=handler`?

Reply:
Mistake made, while resolving conversations. Sorry